mne/time_frequency/psd.py

```diff
@@ -11,17 +11,18 @@
 from .multitaper import _psd_multitaper
 
 
-def _psd_func(epoch, noverlap, nfft, fs, freq_mask, func):
+def _psd_func(epoch, noverlap, n_per_seg, nfft, fs, freq_mask, func):
     """Aux function."""
-    return func(epoch, fs=fs, nperseg=nfft, noverlap=noverlap,
+    return func(epoch, fs=fs, nperseg=n_per_seg, noverlap=noverlap,
                 nfft=nfft, window='hann')[2][..., freq_mask, :]
 
 
-def _check_nfft(n, n_fft, n_overlap):
-    """Helper to make sure n_fft and n_overlap make sense."""
-    n_fft = n if n_fft > n else n_fft
-    n_overlap = n_fft - 1 if n_overlap >= n_fft else n_overlap
-    return n_fft, n_overlap
+def _check_nfft(n, n_fft, n_per_seg, n_overlap):
+    """Helper to make sure n_fft, n_per_seg and n_overlap make sense."""
+    n_per_seg = n_fft if n_per_seg is None or n_per_seg > n_fft else n_per_seg
+    n_per_seg = n if n_per_seg > n else n_per_seg
+    n_overlap = n_per_seg - 1 if n_overlap >= n_per_seg else n_overlap
```

Comment:
I'd rather have people be explicit if they want this behavior. This looks like it's silent-errant-behavior prone to me